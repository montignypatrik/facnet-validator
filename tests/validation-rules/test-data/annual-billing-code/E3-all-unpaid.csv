#;Facture;ID RAMQ;Date de Service;Début;Fin;Periode;Lieu de pratique;Secteur d'activité;Diagnostic;Code;Unités;Rôle;Élément de contexte;Montant Preliminaire;Montant payé;Doctor Info;DEV NOTE - À TRADUIRE: Report_DailyClaimInfo_TotalPrelAmount;DEV NOTE - À TRADUIRE: Report_DailyClaimInfo_TotalFinalAmount;DEV NOTE - À TRADUIRE: Report_DailyClaimInfo_TotalCount;Agence;Patient;Grand Total
1;128357280;15600245860;2025-01-10;08:00;;202520;55489;Aucun Secteur;;15815;0;1;;49,15;0,00;1069491-00000 | Dupont, Marie - Omnipraticien  ;147,45;0,00;3;;ABCD12345678 - Tremblay, Jean;
2;128357281;15600245861;2025-03-15;09:00;;202520;55489;Aucun Secteur;;15815;0;1;;49,15;0,00;1069491-00000 | Dupont, Marie - Omnipraticien  ;147,45;0,00;3;;ABCD12345678 - Tremblay, Jean;
3;128357282;15600245862;2025-06-20;10:00;;202520;55489;Aucun Secteur;;15815;0;1;;49,15;0,00;1069491-00000 | Dupont, Marie - Omnipraticien  ;147,45;0,00;3;;ABCD12345678 - Tremblay, Jean;
