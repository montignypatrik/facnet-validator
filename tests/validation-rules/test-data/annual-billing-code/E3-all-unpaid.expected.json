{
  "scenarioId": "E3",
  "scenarioName": "All Unpaid Billings",
  "description": "Code annuel facturé plusieurs fois, toutes les factures sont non payées",
  "expectedResults": [
    {
      "severity": "error",
      "category": "annual_limit",
      "message": "Le code annuel 15815 a été facturé 3 fois en 2025, toutes les factures sont impayées.",
      "solution": "Veuillez valider la raison du refus et corriger les demandes restantes pour que le tout soit conforme.",
      "ruleData": {
        "monetaryImpact": 49.15,
        "code": "15815",
        "patient": "ABCD12345678",
        "year": 2025,
        "totalCount": 3,
        "paidCount": 0,
        "unpaidCount": 3,
        "dates": ["2025-01-10", "2025-03-15", "2025-06-20"],
        "amounts": [0, 0, 0],
        "totalUnpaidAmount": 147.45,
        "tariffValue": 49.15
      },
      "affectedRecordCount": 3
    },
    {
      "severity": "info",
      "category": "annual_limit",
      "message": "Validation complétée pour 3 enregistrement(s) de codes annuels. 1 erreur(s) détectée(s).",
      "solution": null,
      "ruleData": {
        "monetaryImpact": 49.15,
        "totalRecords": 3,
        "errorCount": 1,
        "totalPotentialGain": 49.15,
        "codesChecked": ["15815"],
        "breakdown": {
          "E1": 0,
          "E2": 0,
          "E3": 1
        }
      }
    }
  ]
}
